components/homme/test_execs/share_kokkos_ut/CMakeLists.txt

```diff
@@ -9,8 +9,8 @@ SET(UTILS_TIMING_DIR ${HOMME_SOURCE_DIR}/utils/cime/CIME/non_py/src/timing)
 # Note: need CUDA_BUILD and HOMMEXX_BFB_TESTING here, since the share
 #       unit tests do not include a config.h file
 SET (COMMON_DEFINITIONS NP=4 NC=4)
-IF (CUDA_BUILD)
-  SET(COMMON_DEFINITIONS ${COMMON_DEFINITIONS} CUDA_BUILD)
+IF (CUDA_BUILD OR HIP_BUILD)
```

Comment:
Could you use HOMMEXX_ENABLE_GPU here and get ride of CUDA/HIP_BUILD entirely?

Reply:
@ambrad  like removing set(HIP_BUILD TRUE) from machine files? i could, but CUDA_BUILD is set in SetCompilerFlags, I do not know why it is done this way. If CUDA_BUILD is removed, what to do with that code?
```
  FIND_PACKAGE(CUDA QUIET)
  IF (${CUDA_FOUND})
    EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
                    OUTPUT_VARIABLE CXX_COMPILER_VERSION_OUT)
    STRING (FIND "${CXX_COMPILER_VERSION_OUT}" "nvcc" pos)
    IF (${pos} GREATER -1)
      SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --expt-extended-lambda")
      SET (CUDA_BUILD TRUE)
```

Reply:
I think all of that code except the last line is still relevant.

Reply:
Ok. More cleanup can be done in a different PR.